### DISCUSSION TITLE
Environment Variables not working when using Vercel CLI

### DISCUSSION THREAD

**Discussion by bjornandrip** [vercel/community#2869] [Help] [answered]
### Summary

When I deploy an application using the Vercel CLI and add my env vars EITHER through the 'vercel env add' command OR in the Vercel UI after the deployment (deployment through CLI) the env variables seem to not work. 

However if I deploy the application through GitHub everything works fine. The environment vars can be used. 

Am I missing something when using the CLI or?
BTW: I have of-course redeployed after setting env variables

### Steps to Reproduce

Deploy an application using the CLI, then set the environment variables and redeploy.
-Does not work

Now do the same thing but through GitHub connection.
-Works

**Comment by amyegan** [comment, marked as answer]
Hi @bjornandrip. I'm not sure if you're building on local and deploying prebuilt via the CLI. If that's the case, make sure you run `vercel pull --environment=preview` before running `vercel build` to ensure that the environment variables are synced with your local environment.

The only other way I could make the environment variable fail is if I didn't prefix it for client side use (`NEXT_PUBLIC_` for my test site). If neither of these scenarios apply to your situation, please walk me through it step by step so I can replicate the problem myself and get a better look at what is going wrong 🙏

**Reply by bjornandrip** [reply]
Hi amyegan. No I am not prebuilding locally the only thing I was doing was typing the "vercel" command to deploy. Now I realise that I should maybe have been deplyoing using "vercel --prod" I did not realise the "vercel" command only deployed to preview. I will try this again with "prod".

**Comment by tungvn** [comment]
I have the same situation.

But when I try to use `vercel cli` to deploy my application to Vercel, it has a problem while building the application.

Actually, in the local environment, it have a `.env.local` file and it has all local env variables. When creating a new build, I see a debug line as the following
`[debug] [2023-06-21T09:39:29.345Z] Loaded environment variables from "/path/to/.vercel/.env.production.local"`
but it didn't use in `.vercel/config.json` and the rewrites rules didn't work as expected.
